app/script/main/app.js

```diff
@@ -756,14 +756,15 @@ z.main.App = class App {
         }
 
         const baseUrl = z.util.Environment.frontend.isLocalhost() ? '/page/auth.html' : '/auth/';
-        let url = `${baseUrl}${location.search}`;
+        let url = `${baseUrl}${window.location.search}`;
         const isImmediateSignOutReason = App.CONFIG.SIGN_OUT_REASONS.IMMEDIATE.includes(signOutReason);
         if (isImmediateSignOutReason) {
           url = z.util.URLUtil.appendParameter(url, `${z.auth.URLParameter.REASON}=${signOutReason}`);
         }
 
         const redirectToLogin = signOutReason !== z.auth.SIGN_OUT_REASON.NOT_SIGNED_IN;
-        if (redirectToLogin) {
+        const forcedAppLogin = z.util.Environment.desktop && window.location.hash === '#login';
```

Comment:
Isn't it supposed to be a || here?
If the app is the desktop or that the current location contains login?

Reply:
Do we really need this dependency between wrapper and webapp? Would be nice to not have to think about this in the future with different versions of stuff here and there.